### PR TITLE
Add multihosting modules

### PR DESCRIPTION
**Pull request by niboo-sam**
Add multihosting modules that allows to use different repository hosting

### CODE REVIEW

**Comment by niboo-sam** [comment]
Hello all,
What about this proposal... Any comments?

**Comment by nhomar** [comment]
Can you extend a little the reader with some print screens to understand
what exactly the module does? The text explaining the feature now is not
clear for me, as far as I can understand what you say it does the actual
set of modules does already.

On Friday, September 11, 2015, Samuel Lefever notifications@github.com
wrote:

> Hello all,
> What about this proposal... Any comments?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/OCA/runbot-addons/pull/60#issuecomment-139546770.

## 

---

Saludos Cordiales

CEO at Vauxoo https://www.vauxoo.com Odoo's Gold Partner.

[image: --]
Nhomar Hernandez
[image: http://]about.me/nhomar
http://about.me/nhomar?promo=email_sig

**Comment by niboo-sam** [comment]
This modules does not add any features for now.
It just changes the structure of runbot repo to be easily connected with other git hosting system.
I have one other module in construction for bitbucket but it could be for any hosting system.
It encapsulates the hosting part so you can add new hosting connector.

File: runbot_github/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{
+    'name': 'Runbot Github Hosting',
+    'category': 'Website',
+    'summary': 'Add the Github hosting to Runbot',
+    'version': '1.0',
```
Comment (osvalr) [inline runbot_github/__openerp__.py]: It must be `8.0.1.0.0`

File: runbot_github/README.rst [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+* (empty)
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/runbot-addons/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/runbot-addons/issues/new?body=module:%20runbot_pylint%0Aversion:%208.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (osvalr) [inline runbot_github/README.rst]: check the version, it must be `8.0.1.0.0`

File: runbot_github/README.rst [hunk 3]
```diff
@@ -0,0 +1,67 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Runbot pylint
```
Comment (elicoidal) [inline runbot_github/README.rst]: @  Thanks for your contribution to the OCA. 
Let me introduce to you some basic information about contribution in OCA and I will review other points in more comments inside the code..
In general, please follow the principles: 
- Contribution guidelines: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md
- Please follow the OCA README template:
  https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst
- We recommend to include a screenshot as well to help new comers (nice to have)
- Please use simple headers in all py files following OCA templates
  https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3
- Please sort all imports in alphabetical order following: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#imports

Here as well  be careful to include a line before the title and it must be same length as the title

File: runbot_github/README.rst [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+===========
+
+This module was written to extend the functionality of runbot to support
```
Comment (elicoidal) [inline runbot_github/README.rst]: This module extends the functionality of runbot to support

File: runbot_github/README.rst [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+Github repository hosting in // of others.
```
Comment (elicoidal) [inline runbot_github/README.rst]: s/"//"/parallel

File: runbot_github/README.rst [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+Usage
+=====
+
+To use this module, you need to:
+
+* Add a repository record and select the right hosting
+
+* Select any needed dependecies among the others repository and precise the version.
+
+For further information, please visit:
```
Comment (elicoidal) [inline runbot_github/README.rst]: remove this part

File: runbot_github/README.rst [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+* https://www.odoo.com/forum/help-1
+
+Known issues / Roadmap
```
Comment (elicoidal) [inline runbot_github/README.rst]: remove this part

File: runbot_github/README.rst (hunk 2 above)
Comment (elicoidal) [inline runbot_github/README.rst]: remove this line following latest README template

File: runbot_github/README.rst [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+This module is maintained by the OCA.
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+To contribute to this module, please visit http://odoo-community.org.
```
Comment (elicoidal) [inline runbot_github/README.rst]: https

File: runbot_github/__init__.py [hunk 9]
```diff
@@ -0,0 +1,23 @@
+# -*- encoding: utf-8 -*-
```
Comment (elicoidal) [inline runbot_github/__init__.py]: Please simplified headers for all py files

File: runbot_github/__openerp__.py [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+    'description': """
```
Comment (elicoidal) [inline runbot_github/__openerp__.py]: remove: already in README

File: runbot_github/__openerp__.py [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+Add Gihub hosting ability to runbot
+
+Contributors
+------------
+* Sylvain Van Hoof
+* Samuel Lefever (sam@niboo.be)
+""",
+    'author': "Eezee-It,Odoo Community Association (OCA)",
+    'depends': ['runbot', 'runbot_multiple_hosting'],
+    'data': [
```
Comment (elicoidal) [inline runbot_github/__openerp__.py]: remove empty keys

File: runbot_github/models/__init__.py [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from . import github
```
Comment (elicoidal) [inline runbot_github/models/__init__.py:4]: protect external imports

File: runbot_github/models/runbot_branch.py [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp import api, models
+from .github import GithubHosting
```
Comment (elicoidal) [inline runbot_github/models/runbot_branch.py]: protect

File: runbot_multiple_hosting/README.rst [hunk 14]
```diff
@@ -0,0 +1,63 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Runbot multiple hosting
```
Comment (elicoidal) [inline runbot_multiple_hosting/README.rst:6]: Pleas one PR per module. 
For README check my previous comments

File: runbot_multiple_hosting/README.rst [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+
+Usage
+=====
+
+To use this module, you need to:
+
+* Select a connector module (Github, Bitbucket,...)
+
+Known issues / Roadmap
+======================
```
Comment (elicoidal) [inline runbot_multiple_hosting/README.rst]: remove

File: runbot_multiple_hosting/README.rst [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+
+* (empty)
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/runbot-addons/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/runbot-addons/issues/new?body=module:%20runbot_pylint%0Aversion:%208.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (elicoidal) [inline runbot_multiple_hosting/README.rst]: remove this line

File: runbot_multiple_hosting/README.rst [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+This module is maintained by the OCA.
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+To contribute to this module, please visit http://odoo-community.org.
```
Comment (elicoidal) [inline runbot_multiple_hosting/README.rst]: https

File: runbot_multiple_hosting/__init__.py [hunk 18]
```diff
@@ -0,0 +1,23 @@
+# -*- encoding: utf-8 -*-
```
Comment (elicoidal) [inline runbot_multiple_hosting/__init__.py]: simplified headers in all py

File: runbot_multiple_hosting/__openerp__.py [hunk 19]
```diff
@@ -0,0 +1,46 @@
...
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{
+    'name': 'Runbot Multiple Hosting',
+    'category': 'Runbot',
+    'summary': 'Offer the base to use several hosting in Runbot',
+    'version': '1.0',
```
Comment (elicoidal) [inline runbot_multiple_hosting/__openerp__.py:9]: 8.0.1.0.0

File: runbot_multiple_hosting/__openerp__.py [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+    'description': """
```
Comment (elicoidal) [inline runbot_multiple_hosting/__openerp__.py]: remove

File: runbot_github/models/__init__.py (hunk 12 above)
Comment (niboo-sam) [reply runbot_github/models/__init__.py:4]: What do you mean?

**Comment by niboo-sam** [comment]
I have corrected what you asked for but I have also rebase the branch.
Should I do a new PR?

**Comment by moylop260** [comment]
You  need create a new pr
Just rebase from 8.0 instead of 9.0

**Comment by niboo-sam** [comment]
I will create a new PR but on V9 as the code has been runned on v9 and I m not sure about the backward compatibility.

**Comment by hbrunn** [comment]
closing due to lack of activity

**Pull request closed without merging**